apps/competitions-api/src/routes/competitions/createCompetition.ts

```diff
@@ -0,0 +1,90 @@
+import { Router } from 'express';
+import { prisma } from '@competition-manager/prisma';
+import { z } from 'zod';
+import { parseRequest, verifyAccessToken } from '@competition-manager/utils';
+import { PAYMENT_METHOD, Competition$ } from '@competition-manager/schemas';
+
+export const router = Router();
+
+const NewCompetition$ = Competition$.pick({
+    name: true,
+    date: true,
+    method: true,
+    publish: true,
+    description: true,
+    oneDayBibStart: true,
+    oneDayPermissions: true,
+});
+
+const Body$ = z.object({
+    name: z.string(),
+    date: z.coerce.date().min(new Date()),
+    method: z.enum(PAYMENT_METHOD),
+    paymentPlanId: z.number(),
+    optionsId: z.array(z.number()).optional(),
+    accessToken: z.string(),
+});
```

Comment:
Je comprends pas pourquoi tu as defini 2 truc différent ici.
Pour moi ça aurait du etre un truc comme ça
```suggestion
const Body$ = Competition$.pick({
    name: true,
    date: true,
    ...
});
```

Reply:
donc tu passerais le paymentPlan et les options en entier pas juste les id ?

Reply:
Ah non il faut juste les id mais alors un truc comme 

```
const Body$ = Competition$.pick({
    name: true,
    date: true,
    ...
}).extend({...})
```
C'est pcq si on change le type de name ou de date il ne faut pas le changer ici aussi je sais pas si tu vois ce que je veux dire

Reply:
ok mais du coup je peux pas supp NewCompetition j'en ai besoin

Reply:
OK je suis d'accord mais je me demande si mettre NewComp dans les shared package est pas mieux.. Même si on l'utilise que ici au moins on garde les schémas au même endroit à voir

Reply:
Mais par contre body doit être NewComp.pick ou .omit